dash.el

```diff
@@ -2166,7 +2166,41 @@ See `-let' for the description of destructuring mechanism."
       ;; TODO: because inputs to the lambda are evaluated only once,
       ;; -let* need not to create the extra bindings to ensure that.
       ;; We should find a way to optimize that.  Not critical however.
-      `(lambda ,(--map (cadr it) inputs)
+      `(lambda ,(-map #'cadr inputs)
+         (-let* ,inputs ,@body))))))
+
+(defmacro -defun (name match-form &rest body)
+  "Return a `defun' NAME which destructures its input as MATCH-FORM and executes BODY.
```

Comment:
"Define NAME as a function which..."

For consistency with the docstring of `defun`, and because a `defun` isn't something that can be "returned" in this context.